Mironov Alexey

m1ralx opened the pull request.

m1ralx committed:
Обязательное задание

m1ralx committed:
Теперь картинки не исчезают при фильтрации, а становятся "неактивными"

m1ralx committed:
Добавил выделение текущего выбора

m1ralx committed:
Добавил картинки с макета. Теперь круто

VorontsovMaxim commented on index.css:
```diff
@@ -1 +1,234 @@
+    overflow: hidden;
+}
+
+.cats .cat:nth-child(n)
```
`.cat:nth-child(n)` - то же самое, что просто `cat`.

VorontsovMaxim commented on index.css:
```diff
@@ -1 +1,234 @@
+    max-width: 91.2%;
+}
+
+.cats .cat:nth-child(4n + 2)
```
:+1:

VorontsovMaxim commented on index.css:
```diff
@@ -1 +1,234 @@
+    display: inline-flex;
+    margin-top: 20px;
+    position: absolute;
+    width: auto;
```
По умолчанию же

VorontsovMaxim commented on index.css:
```diff
@@ -1 +1,234 @@
+    display: flex;
+    background-color: #a9a9a9;
+    padding-top: 10px;
+    padding-bottom: 10px;
```
можно записать одним правилом

VorontsovMaxim commented on index.css:
```diff
@@ -1 +1,234 @@
+    align-content: center;
+    margin-top: 20px;
+    margin-left: 15px;
+    margin-right: 15px;
```
И здесь

VorontsovMaxim commented on index.css:
```diff
@@ -1 +1,234 @@
+    display: flex;
+    height: 20px;
+    border: 1px solid #000;
+    margin-right: 2px;
```
Как-то 2 маловато мне кажется. Все равно выглядит как будто прилипает. Но это так вкусовщина не существенная

VorontsovMaxim commented on index.css:
```diff
@@ -1 +1,234 @@
+    display: flex;
+    width: 100%;
+    align-self: flex-end;
+    justify-content: space-between;
```
:+1:

VorontsovMaxim commented:
На маленьких экранах выглядит как то не ок
![small-screen](https://cloud.githubusercontent.com/assets/14827792/11317796/486dda2c-905d-11e5-8208-124851f7d67d.png)
Можно например ограничить ширину снизу

VorontsovMaxim commented:
Ну и с большими экранами проблема
![big-screen](https://cloud.githubusercontent.com/assets/14827792/11317809/bbd77234-905d-11e5-9214-8e32da894e73.png)
Раскладка не сохраняется

m1ralx committed:
1) Убрал лишние свойства
2) Объединил свойства, где это нужно было
3) Задал minmax ширину для меню с выбором типов и средней части с котиками

m1ralx commented:
@VorontsovMaxim fixed

VorontsovMaxim commented on index.css:
```diff
@@ -1 +1,228 @@
+    max-width: 29%;
+}
+
+.cats .cat:nth-child(n+8)
```
На самом деле, я чего то при первой проверке проглядел. Формула то неправильно будет работать если мы добавим больше картинок.

m1ralx replied:
@VorontsovMaxim Потому что она выше чем следующая?

VorontsovMaxim replied:
По условию если картинок будет больше, то раскладка должна повторяться.
Сейчас так:
- Все картинки - 29%
- `n+8` => 8, 9, 10 ... картинки будут 91.2%
- `4n+2` => 2, 6, 10 ... картинки будут 60.1% 

Применив в таком порядке, получим: 

```
□_□□□
□_□_□
□□□_□
□□□□□
□□□□□
□□□__
□□□□□
...
```

Надеюсь понятно изобразил)
То есть проблема не только с порядком формул, но и вообще с ними самими

m1ralx replied:
@VorontsovMaxim Ааа, там 8n должно быть наверное :) а первая вроде правильная?

VorontsovMaxim replied:
Если просто `4n+2` заменить на `8n+2` то проблему это не решит) Менять надо не только там

m1ralx replied:
@VorontsovMaxim не.
4n+2
8n 
В таком порядке

VorontsovMaxim replied:
Да, так правильно

VorontsovMaxim commented:
Уменьшил окно браузера и чего то уехало
![small-screen](https://cloud.githubusercontent.com/assets/14827792/11319763/01af6cbc-90a4-11e5-9c7e-77dafbdd8b88.png)

VorontsovMaxim commented:
Еще было бы круто, чтобы пустого места 
![2015-11-21_2307](https://cloud.githubusercontent.com/assets/14827792/11319782/bb9f41c4-90a4-11e5-98f7-e6fb62ff2d9d.png)
не было, блок с котами был растянут, а рейтинг прибит справа

m1ralx commented:
@VorontsovMaxim так там же раскладка должна сохраняться. Как себя картинки-то должны вести при этом?

VorontsovMaxim commented:
Лучше всего задать минимальную ширину страницы, чтобы при уменьшении окна, при размере меньше минимального - появлялся горизонтальный скролл. Полностью резиново и универсально все равно не получится сделать. Так что ограничить ширину снизу на мой взгляд - хорошо

m1ralx committed:
Поправил раскладку для больших и маленьких экранов

m1ralx commented:
![image](https://cloud.githubusercontent.com/assets/6783167/11322584/2a4c68de-9115-11e5-82d9-2377755a5ae9.png)
@VorontsovMaxim так сойдёт?

VorontsovMaxim commented:
Да, так уже лучше выглядит. Но мне кажется, что лучше было бы если бы все три колонки были бы по центру, а с боков пустое место
Сейчас я еще вижу, что фотки обрезаются с правого края, кажется, что такого быть не должно)

VorontsovMaxim commented:
И будешь ли ты делать grid?

m1ralx committed:
+space-around
+image size

m1ralx commented:
@VorontsovMaxim grid не буду, изображения и отступы сбоку поправил

m1ralx commented:
@VorontsovMaxim Ааа, неправильно понял про отступы. Сейчас переделаю

m1ralx committed:
justify -> center
убрал пустое место между картинками и рейтинго

VorontsovMaxim commented:
Принято
